ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useMemo, useState } from 'react';
+import { NameType } from '@metamask/name-controller';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { getCurrentChainId } from '../selectors';
+import { fetchNftCollectionsMetadata } from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
+  value: string;
+  type: NameType;
+};
+
+type CollectionsData = {
+  [key: string]: Collection;
+};
+
+export function useNftCollectionsMetadata(
+  requests: UseNftCollectionsMetadataRequest[],
+) {
+  const chainId = useSelector(getCurrentChainId);
```

Comment:
Can we avoid using the global network if possible as we have a pending ticket to support any chain ID provided in the property?

Reply:
Do you think we should fix both this and `useFirstPartyContractNames` hooks in order to use provided `chainId` in this PR or a separate one? Leaving one behind doesn't seem logical, doing both exceeds the scope

Reply:
Just this for now as it's new, but we have a pending ticket to add arbitrary chain support to the petname component.